src/test/suite/didactUriCompletionItemProvider.test.ts

```diff
@@ -215,3 +265,55 @@ async function checkForCommandInList(completions: vscode.CompletionItem[], label
 	}
 	return false;
 }
+
+function delay(ms: number) {
+	return new Promise( resolve => setTimeout(resolve, ms) );
+}
+
+async function executeCompletionTest(input: string, expected: string, selectLastSuggestion? : boolean) {
+	const testFolder = path.resolve(__dirname, '..', '..', '..', './test Fixture with speci@l chars');
+	const filename = path.resolve(testFolder, 'testmy.didact.md');
+	const uri = vscode.Uri.parse(filename);
+	await vscode.workspace.fs.writeFile(uri, Buffer.from(input));
+	const document = await vscode.workspace.openTextDocument(uri);
+	const editor = await vscode.window.showTextDocument(document, vscode.ViewColumn.One, true);
+	await delay(500);
+
+	const newCursorPosition = new vscode.Position(0, input.length);
+	editor.selection = new vscode.Selection(newCursorPosition, newCursorPosition);
+
+	const actualCompletionList = (await vscode.commands.executeCommand(
+		'vscode.executeCompletionItemProvider',
+		document.uri, newCursorPosition)) as vscode.CompletionList;
+	expect(actualCompletionList).to.not.be.null;
+	expect(actualCompletionList.items.length).to.be.at.least(1);
+
+	// if this is available, we have completed the didact://?commandId= part of the completion
+	const startCompletionExists = await checkForCommandInList(actualCompletionList.items, "Start new Didact command link");
+
+	// if this is available we have populated the command list
+	const startCommandCompletionExists = await checkForCommandInList(actualCompletionList.items, "vscode.didact.startDidact");
+
+	// if either is complete, we have expected completions showing up
+	try {
+		expect(startCompletionExists).to.be.true;
+	} catch (e) {
+		expect(startCommandCompletionExists ).to.be.true;		
+	}
```

Comment:
I do not understand these 2 cases. Can you provide more details?

Reply:
I just wanted to make sure that either the didact protocol replacement (didact://?commandId=) was in place or the actual list of command Ids is there. If our completion provider is operating correctly, one of those two cases should be met. If it's not, we have some problems.

Reply:
what do you think to write it `expect(startCompletionExists || startCommandCompletionExists).to.be.true;`?